tools/checklocks/checklocks.go

```diff
@@ -138,8 +152,22 @@ func run(pass *analysis.Pass) (interface{}, error) {
 			continue
 		}
 
+		if Benchmark {
+			start = time.Now()
+		}
+
 		// Check the basic blocks in the function.
 		pc.checkFunction(nil, fn, &lff, nil, false /* force */)
+
+		if Benchmark {
+			end := time.Since(start)
+			pc.pass.ImportPackageFact(pc.pass.Pkg, &pgf)
+			if pgf.FunctionCheckTime == nil {
```

Comment:
If you always allocate FunctionCheckTime when setting up pgf, you can drop checks like this.